library/psa_crypto.c

```diff
@@ -4639,6 +4639,14 @@ psa_status_t psa_crypto_init( void )
 
     /* Initialize the random generator. */
     global_data.entropy_init( &global_data.entropy );
+#if defined(MBEDTLS_PSA_INJECT_ENTROPY)
```

Comment:
I think we should make it conditional on the absence of `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES`, because otherwise in some configurations we add the NVSEED as an entropy source twice. In the best case it means slower operation, extra cost of battery and extra wear on storage, in the worst it might have some unforeseen side effects as well.

Or alternatively we can make the requirement of having `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES` as a prerequisite for `MBEDTLS_PSA_INJECT_ENTROPY` by adding it to `check-config.h`.

Also looking at the current condition in `check-config.h` I would expect either `MBEDTLS_ENTROPY_NV_SEED` or `MBEDTLS_PSA_CRYPTO_STORAGE_C` be sufficient to support `MBEDTLS_PSA_INJECT_ENTROPY` and that we don't need both. Is this the case?

Reply:
`check_config.h` does forbid `MBEDTLS_PSA_INJECT_ENTROPY` together with `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES`. However, I agree with adding `&& !defined(MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES)` here, because 1. not everyone includes `check_config.h` and 2. it's more robust in case we do change `check_config.h`.

We need both `MBEDTLS_ENTROPY_NV_SEED` for the hook to read the seed file and `MBEDTLS_PSA_CRYPTO_STORAGE_C` to write the seed. This is messy and could be simplified, but I don't think it's simple to simplify it. I'd prefer to completely bypass `entropy.c` and use the PSA entropy module, but the PSA entropy module is yet to be written.

Reply:
Thanks, I can see it now, I was looking at the wrong branch.